### ISSUE TITLE
Handle external API errors more gracefully

### ISSUE THREAD

**Issue by ogg1980** [flameshot-org/flameshot#2785]
### Feature Description

When upload to Imgur fails; give the user an option to either try again or save.

Or save the screenshot in "Latest Uploads" and maybe add a try again button?

It's kind of annoying if you've made an elaborate screenshot with arrows, text and whatnot and then click upload and it just errors out.

**Comment by mmahmoudian** [comment]
I agree, this is a good suggestion. At the moment the process can be lossy if the upload doesn't go through.
